File: src/com/aopphp/go/pointcut/TruePointFilter.java
package com.aopphp.go.pointcut;

import com.jetbrains.php.lang.psi.elements.PhpNamedElement;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

/**
 * Canonical PointFilter instance that matches all points.
 */
public final class TruePointFilter implements PointFilter
{
    private static final Set<KindFilter> KIND_ALL = new HashSet<KindFilter>(Arrays.asList(KindFilter.values())) ;

    private static final TruePointFilter INSTANCE = new TruePointFilter();
    private TruePointFilter() {};

    public static PointFilter getInstance()
    {
        return INSTANCE;
    }

    @Override
    public boolean matches(PhpNamedElement element) {
        return true;
    }

    @Override
    public Set<KindFilter> getKind() {
        return KIND_ALL;
    }

    @Override
    public int hashCode() {
        return 0;
    }

    @Override
    public boolean equals(Object obj) {
        return obj instanceof TruePointFilter;
    }
}

File: src/main/kotlin/com/aopphp/go/pointcut/TruePointFilter.kt
package com.aopphp.go.pointcut

import com.jetbrains.php.lang.psi.elements.PhpNamedElement

object TruePointFilter : PointFilter {
    private val KIND_ALL = KindFilter.entries.toSet()

    override fun matches(element: PhpNamedElement) = true
    override fun getKind() = KIND_ALL
    override fun hashCode() = 0
    override fun equals(other: Any?) = other is TruePointFilter
}
